hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/OperationSubsystemStart.java

```diff
@@ -31,35 +36,70 @@
 
     @Override
     public void execute(OperationContext opContext, ModelNode model) throws OperationFailedException {
+
+        final AtomicReference<Thread> newThread = new AtomicReference<>();
+
         try {
             ServiceName name = SubsystemExtension.SERVICE_NAME;
             ServiceRegistry serviceRegistry = opContext.getServiceRegistry(true);
             MonitorService service = (MonitorService) serviceRegistry.getRequiredService(name).getValue();
 
-            boolean restart = model.get("restart").asBoolean(false);
-            if (restart) {
-                LOGGER.debug("Asked to restart the Hawkular Monitor service. Will stop it, then restart it now.");
-                service.stopMonitorService();
-            } else {
-                LOGGER.debug("Asked to start the Hawkular Monitor service");
-            }
-            if (!service.isMonitorServiceStarted()) {
-                Thread newThread = new Thread(new Runnable() {
-                    @Override
-                    public void run() {
-                        service.startMonitorService();
+            final boolean refresh = model.get("refresh").asBoolean(false);
+            final boolean restart = refresh || model.get("restart").asBoolean(false);
+            final long delay = model.get("delay").asLong(0L);
+
+            newThread.set(new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        if (delay > 0) {
+                            Thread.sleep(delay);
+                        }
+
+                        MonitorServiceConfiguration runtimeConfig = null;
+
+                        if (restart) {
+                            if (refresh) {
+                                PathAddress address = WildflyCompatibilityUtils.parseCLIStyleAddress(
+                                        "/subsystem=hawkular-wildfly-agent/");

```

Comment:
you can't do this - hard-coding the address. The address is different when running in domain mode (there is a parent whose name is not known at compile time).